waku/waku_lightpush/protocol.nim

```diff
@@ -35,47 +35,50 @@ type
-  proc handle(conn: Connection, proto: string) {.async, gcsafe, closure.} =
-    let buffer = await conn.readLp(MaxRpcSize.int)
-    let reqDecodeRes = PushRPC.decode(buffer)
-    if reqDecodeRes.isErr():
-      error "failed to decode rpc"
-      waku_lightpush_errors.inc(labelValues = [decodeRpcFailure])
-      return
-
-    let req = reqDecodeRes.get()
-    if req.request.isNone():
+proc handleRequest*(wl: WakuLightPush, peerId: PeerId, buffer: seq[byte]): Future[PushRPC] {.async.} = 
+  let reqDecodeRes = PushRPC.decode(buffer)
+  var
+      isSuccess = false
+      pushResponseInfo = ""
+      requestId = ""
+  if reqDecodeRes.isErr():
+    error "failed to decode rpc", error=decodeRpcFailure
+    waku_lightpush_errors.inc(labelValues = [decodeRpcFailure])
+    pushResponseInfo = decodeRpcFailure
+  else:
+    let pushRpcRequest = reqDecodeRes.get()
+    if pushRpcRequest.request.isNone():
       error "invalid lightpush rpc received", error=emptyRequestBodyFailure
       waku_lightpush_errors.inc(labelValues = [emptyRequestBodyFailure])
-      return
-
-    waku_lightpush_messages.inc(labelValues = ["PushRequest"])
-    let
-      pubSubTopic = req.request.get().pubSubTopic
-      message = req.request.get().message
-    debug "push request", peerId=conn.peerId, requestId=req.requestId, pubsubTopic=pubsubTopic
-
-    var response: PushResponse
-    var handleRes: WakuLightPushResult[void]
-    try:
-      handleRes = await wl.pushHandler(conn.peerId, pubsubTopic, message)
-    except Exception:
-      response = PushResponse(is_success: false, info: some(getCurrentExceptionMsg()))
-      waku_lightpush_errors.inc(labelValues = [messagePushFailure])
-      error "pushed message handling failed", error= getCurrentExceptionMsg()
-
-
-    if handleRes.isOk():
-      response = PushResponse(is_success: true, info: some("OK"))
+      pushResponseInfo = emptyRequestBodyFailure
     else:
-      response = PushResponse(is_success: false, info: some(handleRes.error))
-      waku_lightpush_errors.inc(labelValues = [messagePushFailure])
-      error "pushed message handling failed", error=handleRes.error
+      requestId = pushRpcRequest.requestId
+      let request = pushRpcRequest.request
+      waku_lightpush_messages.inc(labelValues = ["PushRequest"])
+      let
+        pubSubTopic = request.get().pubSubTopic
+        message = request.get().message
+      debug "push request", peerId=peerId, requestId=requestId, pubsubTopic=pubsubTopic
+      var handleRes: WakuLightPushResult[void]
+      try:
+        handleRes = await wl.pushHandler(peerId, pubsubTopic, message)
+        isSuccess = handleRes.isOk()
+        pushResponseInfo = (if isSuccess: "OK" else: handleRes.error)
+      except Exception:
+        pushResponseInfo = getCurrentExceptionMsg()
+
+      if not isSuccess:
+        waku_lightpush_errors.inc(labelValues = [messagePushFailure])
+        error "pushed message handling failed", error=pushResponseInfo
```

Comment:
Minor comment: seems like this can be outdented one level to apply to the entire outer `if...else`. In other words, the error metric increment and log in lines 45, 46 can be removed and error logging done only here.